lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/IMRUBroadcastReduceWithFilePartitionDataSetTest.cs

```diff
@@ -0,0 +1,360 @@
+                    GenericType<IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>>.Class)
+                .Set(REEF.Driver.DriverConfiguration.CustomTraceLevel, TraceLevel.Info.ToString())
+                .Build();
+        }
+
+        /// <summary>
+        /// This test case tests PartitionedDataset Configuration built in this class for IPartitionedInputDataSet and IInputPartition
+        /// </summary>
+        [Fact]
+        public void TestBuildPartitionedDatasetConfiguration()
+        {
+            int count = 0;
+            int numberOfTasks = 4;
+
+            var dataSet = TangFactory.GetTang().NewInjector(BuildPartitionedDatasetConfiguration(numberOfTasks))
+                .GetInstance<IPartitionedInputDataSet>();
+
+            foreach (var partitionDescriptor in dataSet)
+            {
+                var partition =
+                    TangFactory.GetTang()
+                        .NewInjector(partitionDescriptor.GetPartitionConfiguration())
+                        .GetInstance<IInputPartition<IEnumerable<Row>>>();
+
+                using (partition as IDisposable)
+                {
+                    IEnumerable<Row> e = partition.GetPartitionHandle();
+
+                    foreach (var row in e)
+                    {
+                        Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "Data read {0}: ", row.GetValue()));
+                        count++;
+                    }
+                }
+            }
+
+            Assert.Equal(count, numberOfTasks * DataCount);
+        }
+
+        /// <summary>
+        /// Mapper function configuration.
+        /// </summary>
+        /// <returns></returns>
+        protected override IConfiguration BuildMapperFunctionConfig()
+        {
+            return IMRUMapConfiguration<int[], int[]>.ConfigurationModule
+                .Set(IMRUMapConfiguration<int[], int[]>.MapFunction,
+                    GenericType<TestSenderMapFunction<IEnumerable<Row>>>.Class)
+                .Build();
+        }
+
+        /// <summary>
+        /// Test Mapper function. It will access IInputPartition.
+        /// It verifies that the data file has been downloaded to temp folder
+        /// It then iterates the data and verifies the data count. 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        internal sealed class TestSenderMapFunction<T> : IMapFunction<int[], int[]>
+        {
+            private int _iterations;
+
+            [Inject]
+            private TestSenderMapFunction(
+                [Parameter(typeof(TaskConfigurationOptions.Identifier))] string taskId,
+                IInputPartition<T> partition)
+            {
+                var tempFile = TangFactory.GetTang().NewInjector().GetNamedInstance<TempFileFolder, string>();
+                var tmpFileFodler = Directory.GetCurrentDirectory() + tempFile;
+                Assert.True(Directory.Exists(tmpFileFodler));
```

Comment:
Should we also verify here that there is exactly one file in this folder, and it has a name which matches mask "8 random characters 0-9|a-f"? Theoretically if something goes wrong there can be no files or too many files in this folder